lib/grpc/webproxy/GrpcExpressMiddleware.ts

```diff
@@ -0,0 +1,154 @@
+  const router = express.Router();
+  const clients = {};
+  const protos = protoFiles.map(p => include ? grpc.load({ file: p, root: include }) : grpc.load(p));
+  protoFiles
+        .map(p => `${include}/${p}`)
+        .map(p => schema.parse(fs.readFileSync(p)))
+        .forEach((sch, si) => {
+          const pkg = sch.package;
+          if (!sch.services) { return; }
+          sch.services.forEach((s) => {
+            const svc = s.name;
+            const svcarr = getPkg(clients, pkg, true);
+            svcarr[svc] = new (getPkg(protos[si], pkg, false))[svc](grpcLocation, credentials);
+            s.methods.forEach((m) => {
+              if (m.options['google.api.http']) {
+                supportedMethods.forEach((httpMethod) => {
+                  if (m.options['google.api.http'][httpMethod]) {
+                    console.log(colors.green(httpMethod.toUpperCase()), colors.blue(m.options['google.api.http'][httpMethod]));
+                    router[httpMethod](convertUrl(m.options['google.api.http'][httpMethod]), (req, res) => {
+                      const params = convertParams(req, m.options['google.api.http'][httpMethod]);
+                      const meta = convertHeaders(req.headers, grpc);
+                      const ip = req.headers['x-forwarded-for'] || req.connection.remoteAddress;
+                      try {
+                        getPkg(clients, pkg, false)[svc][lowerFirstChar(m.name)](params, meta, (err, ans) => {
+                          // TODO: PRIORITY:MEDIUM - improve error-handling
+                          // TODO: PRIORITY:HIGH - double-check JSON mapping is identical to grpc-gateway
+                          if (err) {
+                            console.error(colors.red(`${svc}.${m.name}`, err.message));
+                            console.trace();
+                            return res.status(500).json({ code: err.code, message: err.message });
+                          }
+                          res.json(convertBody(ans, m.options['google.api.http'].body));
+                        });
+                      } catch (err) {
+                        console.error(colors.red(`${svc}.${m.name}: `, err.message));
+                        console.trace();
+                      }
+                    });
+                  }
+                });
+              }
+            });
+          });
+        });
+  return router;
+};
+
+const getPkg = (client:any, pkg:any, create:boolean = false): Object => {
+  const ls = pkg.split('.');
+  let obj = client;
+  ls.forEach((name) => {
+    if (create) {
+      obj[name] = obj[name] || {};
+    }
+    obj = obj[name];
+  });
+  return obj;
+};
+
+/**
+ * Parse express request params & query into params for grpc client
+ * @param  {Request} req Express request object
+ * @param  {string} url  gRPC url field (ie "/v1/hi/{name}")
+ * @return {Object}      params for gRPC client
```

Comment:
Typedoc doesn't use the types specified in the comments within squiggly brackets since they are inferred, so you can remove the `{Object}` and so on from the comments in this file.